compiler/frontend/src/hir_to_mir.rs

```diff
@@ -268,27 +276,24 @@ impl<'a> LoweringContext<'a> {
                         responsible,
                     );
 
-                    if !pattern.contains_captured_identifiers() {
-                        body.push_reference(nothing)
-                    } else {
-                        // The generated expression will be followed by at least one
-                        // identifier reference, so we don't have to generate a
-                        // reference to the result in here.
-                        self.last_pattern_result = Some((pattern_result, false));
-                        pattern_result
-                    }
+                    self.ongoing_destructuring = Some(OngoingDestructuring {
+                        result: pattern_result,
+                        is_trivial: false,
+                    });
+                    body.push_nothing()
```

Comment:
By moving the `self.ongoing_destructuring = …` above `body.push_if_else(…)`, we can get rid of the `body.push_nothing()`